op-chain-ops/cmd/check-fjord/checks/checks.go

```diff
@@ -0,0 +1,498 @@
+package checks
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/genesis"
+	"github.com/ethereum-optimism/optimism/op-e2e/bindings"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	op_service "github.com/ethereum-optimism/optimism/op-service"
+	"github.com/ethereum-optimism/optimism/op-service/cliapp"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/opio"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+)
+
+func RunApp(args []string) error {
```

Comment:
I think it would be better to move the `RunApp` function into the main package. It only contains logic that is relevant to the main itself and it's probably never gonna be called from other packages.

This would mean exporting more of the `check...` sub-commands but I think that's fine and the right level of abstraction for the split between the public `checks` package and the main package.

The `make...` functions can probably also move with it into `main` in this PR. In a follow-up, when we try to reduce code duplication among check scripts, these could then probably move into a public check script package for reusability.

Reply:
Deleted the `RunApp` function altogether and moved that login inside `main()` since the wrapper is unnecessary if its moved inside the `main` package